homeassistant/components/template/light.py

```diff
@@ -158,19 +158,19 @@ class LightTemplate(TemplateEntity, LightEntity):
 
     def __init__(
         self,
-        hass,
+        hass: HomeAssistant,
         object_id,
         config,
         unique_id,
-    ):
+    ) -> None:
         """Initialize the light."""
         super().__init__(
             hass, config=config, fallback_name=object_id, unique_id=unique_id
         )
         self.entity_id = async_generate_entity_id(
             ENTITY_ID_FORMAT, object_id, hass=hass
         )
-        friendly_name = self._attr_name
+        friendly_name = self._attr_name or self.entity_id
```

Comment:
This seems unrelated to improving how the template light calculates color mode, please move it to a separate PR.